### PR TITLE
Support stats "ratio_in_yjit"

### PR DESCRIPTION
**Pull request by jethrodaniel**
**What does this PR do?**

When ruby's extended YJIT stats are available (`--yjit-stat`), then we have access to numerous additional stats.

This adds `:ratio_in_yjit`, which is only present when additional stats are enabled.

Example:

```shell
RUBYOPT='--yjit --yjit-stats=quiet' ruby -e '
  def foo = "a" * 10
  300.times{foo}
  puts "ratio_in_yjit: #{::RubyVM::YJIT.runtime_stats[:ratio_in_yjit]}"
'
ratio_in_yjit: 0.9550806290358625
```

**Motivation:**

This should allow for a better understanding of how to tune `--yjit-exec-mem-size`.

From https://github.com/ruby/ruby/blob/ef084cc8f4958c1b6e4ead99136631bef6d8ddba/doc/yjit/yjit.md?plain=1#L213-L223:

> If you start Ruby with `--yjit-stats`, e.g. using an environment variable `RUBYOPT=--yjit-stats`,
> `RubyVM::YJIT.runtime_stats[:ratio_in_yjit]` shows the ratio of YJIT-executed instructions in %.
> Ideally, `ratio_in_yjit` should be as large as 99%, and increasing `--yjit-exec-mem-size` often
> helps improving `ratio_in_yjit`.

**Additional Notes:**

Fwiw, I directly based this on the previous YJIT PRs, which were very helpful here:
- https://github.com/DataDog/dd-trace-rb/pull/2711
- https://github.com/DataDog/dd-trace-rb/pull/3661

`ratio_in_yjit` has been available sine 3.3.0: https://github.com/ruby/ruby/blob/caf0d2058aa223515897401ff94e11e1c0671ce0/doc/NEWS/NEWS-3.3.0.md?plain=1#L436

**How to test the change?**

```
RUBYOPT="--yjit --yjit-stats=quiet" bundle exec rake spec:yjit

bundle exec rake spec:yjit # skipped
```

### CODE REVIEW

File: lib/datadog/core/environment/yjit.rb
Comment (ivoanjo) [inline lib/datadog/core/environment/yjit.rb:68]: Minor: Since this is only for the tests to use, I think it's better to keep it there, otherwise it looks like a forgotten/abandoned method (e.g. "why doesn't `available?` call `stats_available?` comes to mind as a question when looking at this without context).

File: .github/workflows/test-yjit.yaml
Comment (jethrodaniel) [inline .github/workflows/test-yjit.yaml:47]: I just copied the job, seemed easier/more maintainable than trying to fit this logic into the matrix using a single job.

Also, I don't see these actions in https://github.com/DataDog/dd-trace-rb/actions, so I'm guessing they need to be approved by a maintainer first.

File: .github/workflows/test-yjit.yaml
Comment (ivoanjo) [inline .github/workflows/test-yjit.yaml:17]:
```suggestion
      RUBYOPT: ${{ matrix.rubyopt }}
```

File: .github/workflows/test-yjit.yaml
Comment (ivoanjo) [inline .github/workflows/test-yjit.yaml:14]:
```suggestion
    name: Test YJIT (${{ matrix.os }}, ${{ matrix.ruby }} ${{ matrix.rubyopt }})
```

File: .github/workflows/test-yjit.yaml
Comment (ivoanjo) [reply .github/workflows/test-yjit.yaml:47]: Yeah, we need to allow it on our side! I'll press the magic button :)

That said, I've suggested a matrix solution for this -- while I don't think it's the best solution every time, it does seem relatively OK on this case (vs having to maintain this whole other config?)

File: spec/datadog/core/runtime/metrics_spec.rb
Comment (ivoanjo) [inline spec/datadog/core/runtime/metrics_spec.rb:1]: Looking again at line 204

```ruby
              skip('Test only runs with YJIT enabled and RubyVM::YJIT.stats_enabled? false')
```

...that doesn't look right -- we can report these metrics regardless of `stats_enabled?` being enabled or not.

Since we're extending the spec, do you mind updating that to remove the `stats_enabled?` reference?

**Review by ivoanjo — COMMENTED** [review]
I've triggered the CI run and left a few notes. TBH I think this could go as-is, but I've suggested a few improvements ;)

File: .github/workflows/test-yjit.yaml
Comment (jethrodaniel) [reply .github/workflows/test-yjit.yaml:13]: Ooh, I like this solution a lot more, many thanks.

**Review by ivoanjo — APPROVED** [review]
:+1: LGTM. I realized my suggestion for the matrix did not address the part where 3.2 doesn't support `--yjit-stats=quiet` so I pushed a tiny change to address that.

I'm just waiting for CI to settle and I'll merge this in. Thanks for the nice contribution! :pray:

File: .github/workflows/test-yjit.yaml
Comment (p-datadog) [inline .github/workflows/test-yjit.yaml:19]: Why is this configuration excluded?

Comment (jethrodaniel) [reply .github/workflows/test-yjit.yaml:19]: `--yjit-stats` exists on ruby 3.2, but it dumps all the stat info after the interpreter exits, and it doesn't include the new metric added (`ratio_in_yjit`, only in 3.3+), so having that for 3.2 isn't valuable here.

`--yjit-stats=quiet` was introduced in 3.3, and _doesn't_ dump that info to stdout.

I can put up a PR with a comment for clarity.

Comment (jethrodaniel) [reply .github/workflows/test-yjit.yaml:19]: for your consideration: https://github.com/DataDog/dd-trace-rb/pull/4010

**Pull request merged**
